src/Interpreters/Streaming/Aggregator.cpp

```diff
@@ -1877,137 +1694,126 @@ void NO_INLINE Aggregator::convertToBlockImplFinal(
 
     if (exception)
         std::rethrow_exception(exception);
+
+    return finalizeBlock(params, getHeader(/* final */ true), std::move(out_cols), /* final */ true, places.size());
 }
 
-template <typename Method, typename Table>
-void NO_INLINE Aggregator::convertToBlockImplNotFinal(
-    Method & method,
-    Table & data,
-    std::vector<IColumn *>  key_columns,
-    AggregateColumnsData & aggregate_columns) const
+template <typename Method, bool use_compiled_functions, AggregateStateType type, typename Table>
```

Comment:
It is a batch process (loop hashmap), so it is added to avoid too many branches during loop processing.
Yes, we can pass it in as a parameter, unless there is a performance problem, and then do another optimization